src/schemas/generator.ts

```diff
@@ -7,107 +7,139 @@ import fs from "fs";
+    "org.hisp.dhis.chart.Series": "any",
+    "org.hisp.dhis.common.DataDimensionItem": "any",
+    "org.hisp.dhis.common.DimensionalObject": "any",
+    "org.hisp.dhis.interpretation.Mention": "any",
+    "org.hisp.dhis.message.Message": "any",
+    "org.hisp.dhis.message.UserMessage": "any",
+    "org.hisp.dhis.period.Period": "any",
+    "org.hisp.dhis.period.RelativePeriods": "any",
+    "org.hisp.dhis.programstagefilter.EventQueryCriteria": "any",
+    "org.hisp.dhis.relationship.RelationshipConstraint": "any",
+    "org.hisp.dhis.relationship.RelationshipItem": "any",
+    "org.hisp.dhis.render.DeviceRenderTypeMap": "any",
+    "org.hisp.dhis.reporttable.ReportParams": "any",
+    "org.hisp.dhis.sms.command.code.SMSCode": "any",
+    "org.hisp.dhis.sms.command.SMSSpecialCharacter": "any",
+    "org.hisp.dhis.trackedentityattributevalue.TrackedEntityAttributeValue": "any",
+    "org.hisp.dhis.trackedentitycomment.TrackedEntityComment": "any",
+    "org.hisp.dhis.trackedentityfilter.EventFilter": "any",
+    "org.hisp.dhis.trackedentityfilter.FilterPeriod": "any",
+    "org.hisp.dhis.trackedentity.TrackedEntityAttributeDimension": "any",
+    "org.hisp.dhis.trackedentity.TrackedEntityProgramOwner": "any",
+    "org.hisp.dhis.common.DimensionalItemObject": "any",
+
     "java.lang.Object": "object",
     "java.util.Map": "object",
 };
 
-const getClassName = (klass: string): string => _.last(klass.split(".")) || "unknown";
+function getModelName(klass: string, suffix?: string): string {
+    const className = _.last(klass.split("."));
+    if (!className) {
+        throw "No class name";
+    } else {
+        return `D2${className}${suffix || ""}`;
+    }
+}
 
-const getType = (property: SchemaProperty): string => {
-    const { fieldName, propertyType, klass, constants, itemPropertyType, itemKlass } = property;
+const getType = (schemas: Schemas, property: SchemaProperty, suffix?: string): string => {
+    const { propertyType, constants, itemPropertyType, itemKlass } = property;
 
     switch (propertyType) {
         case "REFERENCE":
-            return "Ref";
+            return getInterface(schemas, property, suffix);
         case "COLLECTION":
             if (!itemPropertyType || !itemKlass) throw new Error("Missing item info");
 
-            const innerType = getType({
-                fieldName,
-                propertyType: itemPropertyType,
-                klass: itemKlass,
-            });
+            const innerType = getType(
+                schemas,
+                {
+                    ...property,
+                    propertyType: itemPropertyType,
+                    klass: itemKlass,
+                },
+                suffix
+            );
             return `${innerType}[]`;
         case "TEXT":
         case "URL":
         case "PHONENUMBER":
         case "EMAIL":
         case "COLOR":
         case "PASSWORD":
-            return "string";
         case "DATE":
```

Comment:
String for DATE type?

Reply:
While conceptually a date field, it's a plain JS string in the response:

https://play.dhis2.org/2.32.2/api/users/DXyJmlo9rge.json?fields=created

So we would need to do some post-processing to have real `Date`s, but that would mean iterating every data in the response, it would not be cheap.